### PR TITLE
Dictionary alias

### PR DESCRIPTION
**Pull request by khaitruong922**
#569

- Add field `alias` to dictionary settings, which has default value of title
- Make dictionary title editable

### CODE REVIEW

**Comment by khaitruong922** [comment]
**TO DO**: 
- Make popup displays alias instead of title.
- Update preview popup whenever the alias is changed.

**Need advice**
Currently `DOMDataBinder` only binds input element, which does not work for `contenteditable`. Currently my workaround is to create a hidden input field which updates whenever the `contenteditable` element is updated. This is probably not a good pracitce. I'm not too sure of how `DOMDataBinder` class works. Is it possible to add a binding for elements with `contenteditable === "true"`?

**Comment by jamesmaa** [comment]
> Currently DOMDataBinder only binds input element, which does not work for contenteditable. Currently my workaround is to create a hidden input field which updates whenever the contenteditable element is updated. This is probably not a good pracitce. I'm not too sure of how DOMDataBinder class works. Is it possible to add a binding for elements with contenteditable === "true"?

@Kuuuube could you help out with this if you happen to know something?

**Comment by Kuuuube** [comment]
From a quick look I dont see any reason why adding a div or other contenteditable element is an issue. DOMDataBinder doesnt seem particularly complex either.

Creating a hidden element to mirror the setting sounds like a future hell-bug waiting to happen.

**Comment by khaitruong922** [comment]
Should we use dictionary alias or original dictionary name in Anki `glossary` field?

**Comment by Casheeew** [comment]
> Should we use dictionary alias or original dictionary name in Anki `glossary` field?

I believe so. We should only use the original dictionary name for keying purposes (I believe that's how it's implemented currently?) but basically use the dictionary alias everywhere else for consistency.

**Comment by YukiNagat0** [comment]
> Should we use dictionary alias or original dictionary name in Anki `glossary` field?



> I believe so.

I, actually, think, that you **should not change** behavior of interaction with Anki.
My reasoning is that this can break note types like `jp-mining-note`, where you specify which dictionaries are bilingual and which monolingual (using regex and `glossary`).
Also this adds questions like "What to do if alias is empty-string?", "Should we then use original dictionary name?"

I encourage you not to change Anki integration behavior.

As example of what can happen: In latest update `"selectionText"` was changed to `"popupSelectionText"` what resulted in inability to use the selection of dictionary functionality.

**Comment by YukiNagat0** [comment]
Let's keep it just a visual change **within** the Yomitan.

File: test/data/templates/old-default-anki-field-templates.handlebars [hunk 1]
```diff
@@ -28,6 +28,10 @@
     {{~definition.dictionary~}}
 {{/inline}}
 
+{{#*inline "dictionary-alias"}}
```
Comment (StefanVukovic99) [inline test/data/templates/old-default-anki-field-templates.handlebars:31]: This file shouldn't be edited, it's there just to test that applying all the handlebars updates to it results in the latest default handlebars file.

File: test/data/anki-note-builder-test-results.json [hunk 2]
```diff
@@ -80,6 +82,7 @@
         "cloze-suffix": "cloze-suffix",
         "conjugation": "",
         "dictionary": "Test Dictionary 2",
+        "dictionary-alias": "termsDictAlias",
         "document-title": "title",
         "expression": "打",
         "frequencies": "<ul style=\"text-align: left;\"><li>Test Dictionary 2: 1</li><li>Test Dictionary 2: four</li><li>Test Dictionary 2: five (5)</li><li>Test Dictionary 2: 8</li><li>Test Dictionary 2: fourteen</li><li>Test Dictionary 2: twenty (20)</li><li>Test Dictionary 2: 26</li></ul>",
```
Comment (StefanVukovic99) [inline test/data/anki-note-builder-test-results.json:88]: should the frequencies handlebar also use alias?

File: ext/js/background/backend.js [hunk 3]
```diff
@@ -2472,6 +2472,7 @@ export class Backend {
         if (mode === 'merge' && !enabledDictionaryMap.has(mainDictionary)) {
             enabledDictionaryMap.set(mainDictionary, {
                 index: enabledDictionaryMap.size,
+                alias: mainDictionary,
```
Comment (StefanVukovic99) [inline ext/js/background/backend.js:2475]: Should this maybe be the mainDictionary's alias instead of it's name?

Comment (khaitruong922) [reply ext/js/background/backend.js:2475]: https://github.com/FooSoft/yomichan/pull/1729

Looks like this is used for grouping using main dictionary, but excluded all the results from the main dictionary. Basically, the main dictionary here will never be shown on popup so the alias does not matter here.

**Review by StefanVukovic99 — APPROVED** [review]

**Review by Kuuuube — CHANGES_REQUESTED** [review]
Can this be fixed? When clicking on empty space next to the dict name it starts editing the name of the dict below it something like this:
image

**Comment by Kuuuube** [comment]
Not seeing any way to revert the alias also. There should be a button or something somewhere to revert it. Maybe on the kebab menu.

**Comment by khaitruong922** [comment]
The user can empty the alias to reset it back to the original name. Adding an option on kebab menu sounds fine.

**Comment by khaitruong922** [comment]
Move alias edit inside modal which is opened kebab menu and use `input` instead.
- Remove `contenteditable` and add `element` type in `dom-data-binder`
- Add Reset button to set the input to dictionary name.

**Review by StefanVukovic99 — APPROVED** [review]
I think it's good. 
I'd maybe like to have the `Set Alias` be called `Rename` to be easier to understand. The `{dictionary}` and `{dictionary-alias}` handlebars might then be less understandable, but I don't they get much use anyway.

**Comment by khaitruong922** [comment]
I would prefer something in between like changing the menu option to `Rename` but the help message will be kept as `Input the alias for X dictionary:`. It should be clear that this is not a complete rename (the user cannot rename then add the same dictionary) but just a visual change for pop-up and Anki.

`

**Comment by StefanVukovic99** [comment]
The term `Display name` might also be helpful somewhere

**Comment by khaitruong922** [comment]
Documentation and menu updated with new texts. Still keeps the `alias` for code naming and Anki marker,

**Review by Kuuuube — APPROVED** [review]

**Pull request merged**